elements/dash-tree/dash-tree.js

```diff
@@ -0,0 +1,405 @@
+      const activeItem = this.querySelector('.active');
+      if (activeItem)
+        activeItem.classList.remove('active');
+      // If the element is expandable then it should have a <span> label
+      // as an immediate child. Put the `active` class on that child instead.
+      if (treeItem.hasAttribute('aria-expanded')) {
+        treeItem.firstElementChild.classList.add('active');
+        return;
+      }
+      treeItem.classList.add('active');
+    }
+
+    /**
+     * If focus is on an open node, closes the node.
+     * If focus is on a child node that is also either an end node or a closed
+     * node, move focus to its parent node.
+     */
+    _collapseTreeItem() {
+      const treeItem = this._currentTreeItem();
+      if (treeItem.firstElementChild && isExpanded(treeItem)) {
+        treeItem.setAttribute('aria-expanded', 'false');
+        return;
+      }
+      // Walk up the tree till you find the parent `<dash-treeitem>`.
+      // If this is a root node, do nothing. Otherwise, collapse the parent
+      // and move focus to it.
+      let parent = treeItem.parentElement;
+      if (parent === this)
+        return;
+      while (!isTreeItem(parent)) {
+        parent = parent.parentElement;
+      }
+      parent.setAttribute('aria-expanded', 'false');
+      this._focusTreeItem(parent);
+    }
+
+    _expandTreeItem() {
+      const treeItem = this._currentTreeItem();
+      if (treeItem.firstElementChild) {
+        treeItem.setAttribute('aria-expanded', 'true');
+      }
+    }
+
+    /**
+     * Perform the default action for a treeitem. If the item is a parent
+     * node, toggle its expanded/collapsed state. If the item is an end
+     * node, dispatch an event with a reference to the node. If this was
+     * a file picker, an application could listen for this event and open
+     * the file based on the item's name.
+     */
+    _selectTreeItem() {
+      // Set the item to aria-selected and remove aria-selected from
+      // any other nodes
+      const treeItem = this._currentTreeItem();
+      const treeItems = this.querySelectorAll('dash-treeitem');
+      for (let i = 0; i < treeItems.length; i++) {
+        treeItems[i].removeAttribute('aria-selected');
+      }
+      treeItem.setAttribute('aria-selected', 'true');
+
+      // There can be only one selected item at a time.
+      // Find any previous selected item and remove its selected class.
+      const selectedItem = this.querySelector('.selected');
+      if (selectedItem)
+        selectedItem.classList.remove('selected');
+
+      if (treeItem.hasAttribute('aria-expanded')) {
+        if (isExpanded(treeItem)) {
+          this._collapseTreeItem();
+        } else {
+          this._expandTreeItem();
+        }
+        // Since this is an expandable element, set the selected class
+        // on its `<span>` label.
+        treeItem.firstElementChild.classList.add('selected');
```

Comment:
So this branch doesn’t send a `dash-tree-item-selected` event. Is that intentional?